src/components/AgentAuthority/ContentWrap/ContentWrap.js

```diff
@@ -6,9 +6,14 @@
 import ColumnLeftWrap from '../ColumnLeft/ColumnLeftWrap'
 import ColumnRightWrap from '../ColumnRight/ColumnRightWrap'
 
+import RequestProof from '../RequestProof'
+import IssueLicense from '../IssueLicense'
+
 export default function ContentWrap({ origin }) {
   return (
     <>
+      <RequestProof origin={origin} />
+      <IssueLicense origin={origin} />
```

Comment:
This is pretty yucky. So each of these components (and their subcomponents) is handling a different part of the flow on loops in a disconnected way. And then they each render their own Errors completely separate from one another right? I wonder if we could handle this better in a single hook that then did the whole flow and reported an amalgameted error/status back up?

Reply:
Yep `RequestProof` polls for proposals and `IssueLicense` polls for verified proofs

Reply:
changed to mostly use a single hook (apart from polling for proposals) with callbacks